pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1219,31 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	p.log.Info("requesting ports to delete")
+	port_list, err = ports.List(
+		p.client,
+		ports.ListOpts{NodeUUID: p.status.ID},
+	).Extract()
+	switch err.(type) {
+	case nil:
+		p.log.Info("port list obtained")
+	default:
+		return result, errors.Wrap(err, "failed to obtain port list")
+	}
+
+	for port_to_delete := range port_list {
+		err = port.Delete(p.client, port_to_delete.UUID).ExtractErr()	
+		switch err.(type) {
+		case nil:
+			p.log.Info("port removed")
+		case gophercloud.ErrDefault409:
+			p.log.info("could not remove port, host locked")
```

Comment:
The returned error will be logged, so there's no need to log a message here.

Reply:
Good point, thanks!